cpp/src/umap/simpl_set_embed/optimize_batch_kernel.cuh

```diff
@@ -38,6 +39,17 @@ DI T2 rdist(const T *X, const T *Y, int n) {
   }
   return result;
 }
+template <typename T, typename T2, int LEN>
+DI T2 rdist(const T (&X)[LEN], const T (&Y)[LEN]) {
+  auto result = T2(0.0);
+#pragma unroll
```

Comment:
Is the unroll only affected in the cases where the arrays being manipulated in the loop are registers?

Reply:
ideally compiler should have figured out the loop is compile constant based and thus unroll this loop automatically. However, somehow compiler wasn't doing this as expected (which was thereby causing register spills!). Thus, I explicitly added unroll hint.